src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -131,7 +131,9 @@ internal sealed partial class TexlBinding
         /// <summary>
         /// Default name used to access a Lambda scope.
         /// </summary>
-        internal DName ThisRecordDefaultName => new DName("ThisRecord");
+        internal static DName ThisRecordDefaultName => new DName("ThisRecord");
+
+        internal static DName ThisItemDefaultName => new DName("ThisItem");
```

Comment:
>ThisItem [](http://example.com/codeflow?start=64&length=8)

Would this conflict with gallery ThisItem when used inside a gallery context?

Reply:
Yes it would. Greg asked for ThisItem. If you wanted to use a gallery ThisItem, you would need to use the Table function and the regular ForAll.

Reply:
I guess you can also use With outside of ForAll to store ThisItem value.

With({Item: ThisItem}, ForAll(...